testcafe/tests/pages/drive-model.js

```diff
@@ -0,0 +1,83 @@
+import { Selector, t } from 'testcafe'
+import {
+  getElementWithTestId,
+  getPageUrl,
+  checkExistVisible
+} from '../helpers/utils'
+
+export default class Page {
+  constructor() {
+    //Sidebar
+    this.sidebar = getElementWithTestId('driveSidebar')
+    this.btnNavToFolder = getElementWithTestId('navToFolder')
+    this.btnNavToRecent = getElementWithTestId('navToRecent')
+    this.btnNavToSharing = getElementWithTestId('navToSharing')
+    this.btnNavToTrash = getElementWithTestId('navToTrash')
+    this.breadcrumbTitle = getElementWithTestId('path-title')
+    //Main menu
+    this.cozBar = Selector('#coz-bar')
+    this.btnMainApp = this.cozBar.find('div.coz-nav-apps-btns > button')
+    this.cozNavPopContent = Selector('[class*=coz-nav-pop]')
+    this.btnCozBarDrive = this.cozNavPopContent
+      .find('li.coz-nav-apps-item > a')
+      .withText('Cozy Drive') // !FIXME: do not use text
+
+    //Error and empty folders
+    // !FIXME better selector for error
+
+    //c-empty is use for empty drive, or error..
+    this.driveEmpty = Selector('[class*="c-empty"]')
+      .parent(0)
+      .withAttribute('data-test-id', 'fil-content-body')
+
+    //oops is a class for error only
+    this.errorOops = Selector('[class*="oops"]')
+
+    this.errorEmpty = Selector('[class*="c-empty"]')
+      .child('h2')
+      .withText('Switch online!') // !FIXME: do not use text
+  }
+
+  async checkSidebarButton(btn, path, title) {
+    checkExistVisible(btn, `Button ${title}`)
+
+    await t.expect(btn.getAttribute('href')).contains(path)
+    await this.navigateToPage(btn, path, title)
```

Comment:
Why a method named `checkSidebarButton` call a method `navigateToPage` method?

Reply:
Because it check the button is fully functionnal, inclunding that the button can be use to navigate to some page. But I agree this name is kinda confusing. maybe something like `clickAndNavigate` ?